school/src/test/java/ru/hogwarts/school/StudentControllerWebMvcTest.java

```diff
@@ -0,0 +1,207 @@
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        when(studentRepository.findById(id)).thenReturn(Optional.of(student));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/" + id)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.age").value(age));
+
+    }
+    @Test
+    public void deleteStudentTest() throws Exception {
+
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        doNothing().when(studentRepository).deleteById(id);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .delete("/student/" + id))
+                .andExpect(status().isOk());
+    }
+    @Test
+    public void findByAgeTest() throws Exception {
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        when(studentRepository.findByAge(age)).thenReturn(List.of(student));
+
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.age").value(age));
+    }
+
+    @Test
+    public void findByAgeBetweenTest() throws Exception {
+
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        doNothing().when(studentRepository).findByAgeBetween(1, 10);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/" + 1 + 10))
+                .andExpect(status().isOk());
```

Comment:
```suggestion
        when(studentRepository.findByAgeBetween(0,10)).thenReturn(List.of(student));

        mockMvc.perform(MockMvcRequestBuilders
                        .get("/student?min=1&max=10"))
                .andExpect(status().isOk());
```
Неправильно мокаешь метод и указываешь урл